pennylane/pauli/utils.py

```diff
@@ -140,24 +140,24 @@ def are_identical_pauli_words(pauli_1, pauli_2):
     # pylint: disable=isinstance-second-argument-not-valid-type
     """Performs a check if two Pauli words have the same ``wires`` and ``name`` attributes.
 
-    This is a convenience function that checks if two given :class:`~.Tensor` instances specify the same
-    Pauli word. This function only checks if both :class:`~.Tensor` instances have the same wires and name
-    attributes, and hence won't perform any simplification to identify if the two Pauli words are
-    algebraically equivalent. For instance, this function will not identify
-    that ``PauliX(0) @ PauliX(0) = Identity(0)``, or ``PauliX(0) @ Identity(1)
-    = PauliX(0)``, or ``Identity(0) = Identity(1)``, etc.
+    This is a convenience function that checks if two given :class:`~.Tensor` or :class:`~.Prod`
+    instances specify the same Pauli word. This function only checks if both :class:`~.Tensor`
+    or :class:`~.Prod` instances have the same wires and name attributes, and hence won't perform
+    any simplification to identify if the two Pauli words are algebraically equivalent. For
+    instance, this function will not identify that ``PauliX(0) @ PauliX(0) = Identity(0)``, or
+    ``PauliX(0) @ Identity(1) = PauliX(0)``, or ``Identity(0) = Identity(1)``, etc.
```

Comment:
Actually, it does now:
```
qml.operation.enable_new_opmath()
X, Y, Z = qml.PauliX, qml.PauliY, qml.PauliZ
op1 = X(0)
op2 = X(0) @ qml.Identity(1)
>>> qml.pauli.are_identical_pauli_words(op1, op2)
True
```

Also this works:
```
op1 = qml.ops.Sum(X(0), X(0))
op2 = X(0) @ qml.Identity(1)
>>> qml.pauli.are_identical_pauli_words(op1, op2)
True
```

So getting a free upgrade 👨‍🍳 👌

Reply:
This could actually come in handy for some of the DLA stuff where we want exactly this check (e.g. 2 * X0 == X0)